beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -444,9 +461,10 @@ func (f *ForkChoice) UpdateJustifiedCheckpoint(jc *forkchoicetypes.Checkpoint) e
 	}
 	f.store.checkpointsLock.Lock()
 	defer f.store.checkpointsLock.Unlock()
+	f.store.prevJustifiedCheckpoint = f.store.justifiedCheckpoint
 	f.store.justifiedCheckpoint = jc
 	bj := f.store.bestJustifiedCheckpoint
-	if bj == nil || jc.Epoch > bj.Epoch {
+	if bj == nil || bj.Root == params.BeaconConfig().ZeroHash || jc.Epoch > bj.Epoch {
```

Comment:
These setters are only used at launch, we should consider moving these checkpoints to `New()`. It is not done in this PR because of the extra large number of tests that would fail from such a change.